test-case/check-fw-echo-reference.sh

```diff
@@ -84,17 +75,20 @@ do
         fi
         # run echo reference test
         dlogc "alsabat -P $pb_dev --standalone -c $channel -f $fmt -r $rate -n $frames -F $frequency"
-        alsabat -P "$pb_dev" --standalone -c "$channel" -f "$fmt" -r "$rate" -n "$frames" -F "$frequency" &
+        alsabat -P "$pb_dev" --standalone -c "$channel" -f "$fmt" -r "$rate" -n "$frames" -F "$frequency" & alsabatPID=$!
         # playback may have low latency, add 0.5 second delay to aviod recording zero at beginning.
         sleep 1
         dlogc "alsabat -C $cp_dev -c $channel -f $fmt -r $rate -F $frequency"
-        alsabat -C "$cp_dev" -c "$channel" -f "$fmt" -r "$rate" -F "$frequency"
+        if ! alsabat -C "$cp_dev" -c "$channel" -f "$fmt" -r "$rate" -F "$frequency"; then
+            # upload failed wav files
+            find /tmp -maxdepth 1 -type f -name "bat.wav.*" -size +0 -exec cp {} "$LOG_ROOT/" \;
+            exit 1
+        fi
 
-        # upload failed wav file
-        if [[ "$?" != "0" ]]; then
-            __upload_wav_file
-        exit 1
+        if ! timeout 2 bash -c wait $alsabatPID; then
```

Comment:
I tried timeout 2 wait $pid directly, but failed with "timeout: failed to run command ‘wait’: No such file or directory"
but looks like with bash -c, it can only fix the error, but it cannot return error code when $pid is timed out.
so I changed the way to check it.